### PR TITLE
Migration: do not add duplicate projects

### LINKED ISSUE #5694: dotnet migrate is adding a duplicate folder in the SLN file.
## Steps to reproduce
1. git clone git@github.com:exceptionless/Exceptionless.Net.git
2. Edit global.json by removing the SDK entry from it.
3. Run dotnet migrate

## Expected  behavior
The updated Exceptionless.Net.sln file is correct.

## Actual behavior
It has two entries for Samples.

## Environment data
`dotnet --info` output:
1.0.0-rc4-004771

### PR DESCRIPTION
**Pull request by jgoshi**
Fixes #5694 

Duplicate solution folder projects were being added if the existing solution folder had different casing than the on-disk folder. Duplicate projects were being added if the existing solution folder had file paths with escaped directory separators.

@piotrpMSFT @livarcocc @krwq @jonsequitur @eerhardt

### CODE REVIEW

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs [hunk 1]
```diff
@@ -261,6 +261,32 @@ public static string GetPathWithDirectorySeparator(string path)
...
+            {
+                if (!string.IsNullOrEmpty(component))
+                {
+                    result = Path.Combine(result, component);
+                }
+            }
+
+            if (path[path.Length-1] == Path.DirectorySeparatorChar)
+            {
+                result += Path.DirectorySeparatorChar;
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: You are adding an extra path separator at the end of the string?

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs [hunk 2]
```diff
@@ -261,6 +261,32 @@ public static string GetPathWithDirectorySeparator(string path)
...
         }
 
+        public static string RemoveExtraPathSeparators(string path)
+        {
+            if (string.IsNullOrEmpty(path))
+            {
+                return path;
+            }
+
+            var components = path.Split(Path.DirectorySeparatorChar);
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: Do the paths in VS always use the path separator according to the OS?

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs (hunk 1 above)
Comment (jgoshi) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: If the trailing path separator was there in the input string I was thinking we should keep it.

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs (hunk 2 above)
Comment (jgoshi) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: Not necessarily. But in the FilePath api (in SlnFile) we always convert to OS specific path separator. This utility method shouldn't know about VS so I just made it always deal in OS specific paths.

Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: Awesome. I thought so but wanted to confirm.

File: src/Microsoft.DotNet.Cli.Utils/PathUtility.cs (hunk 1 above)
Comment (livarcocc) [inline src/Microsoft.DotNet.Cli.Utils/PathUtility.cs]: Ah, I see, we check the path and add it to result, I got confused between the two variables.

**Comment by livarcocc** [comment]
@dotnet-bot Test CentOS7.1 x64 Debug Build

**Comment by livarcocc** [comment]
@dotnet-bot Test OSX x64 Release Build

**Pull request merged**